### PR TITLE
Bug problem with asset register

### LINKED ISSUE #6842: Problem with fixed asset register
We have noticed that when adjusting assets, transferring and removing stock as a loss, the asset register continues to display assets whose quantity in stock is equal to Zero, this issue proposes to solve this problem.

image

### PR DESCRIPTION
**Pull request by lomamech**
- Hide the asset where quantity in stock is 0
- Add HAVING filters in SQL Clause format
**_Bonus_**
- Set Serial Number not required in lot entry stock
- Dont display expiry information for non Expirable asset

closes https://github.com/IMA-WorldHealth/bhima/issues/6842

### CODE REVIEW

File: server/controllers/stock/core.js [hunk 1]
```diff
@@ -318,6 +319,8 @@ async function getAssets(params) {
   }
   filters.setGroup(groupByClause);
 
+  filters.setHaving(havingClause);
```
Comment (jmcameron) [inline server/controllers/stock/core.js:322]: 👍

File: server/lib/filter.js [hunk 2]
```diff
@@ -215,8 +228,9 @@ class FilterParser {
     const conditionStatements = this._parseStatements();
     const order = this._order;
     const group = this._group;
+    const having = this._having;
 
-    return `${sql} WHERE ${conditionStatements} ${group} ${order} ${limitCondition}`;
+    return `${sql} WHERE ${conditionStatements} ${group} ${having}${order} ${limitCondition}`;
```
Comment (jmcameron) [inline server/lib/filter.js]: I think you may want to add a space between the ${having} and ${order} so the developer does not need to add a space when using both.

**Review by jmcameron — COMMENTED** [review]
I looked over the code and pointed out a minor issue.

I tested (with bhima_test) and it all seems work work fine.

However, when I looked at the articles in stock and hovered over the warning triangle for Quinine, I noticed that the tooltip had not been fully translated.  See:
image

This can be fixed by editing client/src/modules/stock/inventories/registry.js and adding a line to translate the unit_type after line 275 so it looks like this:

        // set status flags
        rows.forEach(row => {
          setStatusFlag(row);
          row.status_translated = $translate.instant(Stock.statusLabelMap(row.status));
          row.unit_type = $translate.instant(row.unit_type);
        });

**Comment by mbayopanda** [comment]
@lomamech could you consider @jmcameron suggestions

**Comment by lomamech** [comment]
@mbayopanda @jmcameron  all checks have passed

**Pull request merged**